dbgpt/storage/knowledge_graph/community_summary.py

```diff
@@ -158,24 +318,66 @@ async def asimilar_search_with_scores(
         filters: Optional[MetadataFilters] = None,
     ) -> List[Chunk]:
         """Retrieve relevant community summaries."""
-        # global search: retrieve relevant community summaries
+        # Global search: retrieve relevant community summaries
         communities = await self._community_store.search_communities(text)
         summaries = [
             f"Section {i + 1}:\n{community.summary}"
             for i, community in enumerate(communities)
         ]
         context = "\n".join(summaries) if summaries else ""
 
-        # local search: extract keywords and explore subgraph
         keywords = await self._keyword_extractor.extract(text)
-        subgraph = self._graph_store.explore(keywords, limit=topk).format()
-        logger.info(f"Search subgraph from {len(keywords)} keywords")
 
-        if not summaries and not subgraph:
+        # Local search: extract keywords and explore subgraph
+        subgraph = MemoryGraph()
+        subgraph_for_doc = MemoryGraph()
+
+        if os.getenv("ENABLE_KNOWLEDGE_GRAPH_SEARCH").lower() == "true":
+            subgraph = self._graph_store_apdater.explore(
+                subs=keywords, limit=topk, search_method="entity_search"
+            )
+
+            if os.getenv("ENABLE_DOCUMENT_GRAPH_SEARCH").lower() == "true":
+                keywords_for_chunk_search: List[str] = []
+                for vertex in subgraph.vertices():
+                    keywords_for_chunk_search.append(vertex.name)
+
+                subgraph_for_doc = self._graph_store_apdater.explore(
```

Comment:
I suggest splitting it into two steps: find the one-hop chunk of the entity. Here, the number of chunks with content can be limited by limit. Then, retrieve the complete upstream directory path based on the chunks after limit to ensure the integrity of the document structure.

Reply:
done